src/commands/coffeechats/coffeeMatch.ts

```diff
@@ -0,0 +1,50 @@
+import { Message } from 'discord.js';
+import { CommandoClient, CommandoMessage } from 'discord.js-commando';
+import { writeHistoricMatches, getMatch } from '../../components/coffeechat';
+import { AdminCommand } from '../../utils/commands';
+import _ from 'lodash';
+
+class coffeeMatchCommand extends AdminCommand {
+  constructor(client: CommandoClient) {
+    super(client, {
+      name: 'coffee-match',
+      aliases: ['coffeematch'],
+      group: 'coffeechats',
+      memberName: 'match',
+      description: 'Matches current coffee chat roles',
+      examples: [`${client.commandPrefix}coffeematch`]
+    });
+  }
+
+  async onRun(message: CommandoMessage): Promise<Message> {
+    //makes sure future matches are valid (made for the current group / still has matches left)
+    const matches = await getMatch(this.client);
+    await this.alertMatches(matches);
+    await writeHistoricMatches(matches);
+    return message.reply(`Sent ${matches.length} match(es).`);
+  }
+
+  alertMatches = async (matches: string[][]): Promise<void> => {
+    const outputMap: Map<string, string[]> = new Map();
+    //map them to find what to send a specific person
+    matches.forEach((pair) => {
+      if (!outputMap.get(pair[0])) outputMap.set(pair[0], []);
+      if (!outputMap.get(pair[1])) outputMap.set(pair[1], []);
+      outputMap.get(pair[0])!.push(pair[1]);
+      outputMap.get(pair[1])!.push(pair[0]);
+    });
+    //send out messages
+    outputMap.forEach(async (targets, user) => {
+      const discordUser = await this.client.users.fetch(user);
+      //we use raw discord id ping format to minimize fetch numbers on our end
```

Comment:
how does using discord ID ping format minimize fetches, is the alternative just using the `discordUser` object?

Reply:
basically just saves us an await call, since clients.users.fetch is async.

Alternatively, we would have to build a dict of all the id's so that we don't have multiple instances of the same discord User in the map.

Reply:
oh I see what this is doing, I think this is fine as long as the @mention won't just show up as an ID string as that's bad user experience

Reply:
For some reason this seems to show ID's occasionally, and I can't figure out whether its a discord caching issue or something else entirely. I think a safer bet is to just refactor this to use the fetch properly.